src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -468,27 +463,27 @@ internal void ParseSolutionFileForConversion()
         internal void ParseSolutionFile()
         {
             ErrorUtilities.VerifyThrow(!string.IsNullOrEmpty(_solutionFile), "ParseSolutionFile() got a null solution file!");
-            ErrorUtilities.VerifyThrowInternalRooted(_solutionFile);
+            ErrorUtilities.VerifyThrowInternalRooted(_solutionFile!);
 
-            FileStream fileStream = null;
             SolutionReader = null;
 
             try
             {
-                // Open the file
-                fileStream = File.OpenRead(_solutionFile);
-                SolutionReader = new StreamReader(fileStream, Encoding.GetEncoding(0)); // HIGHCHAR: If solution files have no byte-order marks, then assume ANSI rather than ASCII.
+                // Open the file.
+                using FileStream fileStream = File.OpenRead(_solutionFile!);
+
+                SolutionReader = new StreamLineSpanReader(
+                    fileStream,
```

Comment:
Oh, so they don't rewind the stream after opening it for the version?

Reply:
Assuming you're referring to `GetSolutionFileAndVisualStudioMajorVersions`, that's right. That method only accepts a file path, and creates its own stream/reader/etc. I created an overload that accepts a `TextReader` to call from unit tests.

Looking deeper I think you're right. It would be possible to change `ProjectInstance.LoadSolutionForBuild` such that it sniffs the header, then rewinds the stream and parses the full file again once it calls `GenerateSolutionWrapper`.

Would you expect this to make much difference? I don't see any direct coverage of these methods from unit tests (I'm sure they're covered elsewhere) so am reluctant to change it in this PR.

Reply:
Yeah I wouldn't change it, just thought it was a bit odd.